csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -75,6 +75,30 @@ public static SessionOptions MakeSessionOptionWithCudaProvider(int deviceId = 0)
             return options;
         }
 
+        /// <summary>
+        /// A helper method to construct a SessionOptions object for CUDA execution.
+        /// Use only if CUDA is installed and you have the onnxruntime package specific to this Execution Provider.
+        /// </summary>
+        /// <param name="cuda_options">CUDA EP provider options to configure the CUDA EP instance</param>>
+        /// <returns>A SessionsOptions() object configured for execution with CUDA provider options.</returns>
+        public static SessionOptions MakeSessionOptionWithCudaProvider(OrtCUDAProviderOptions cuda_options)
+        {
+            CheckCudaExecutionProviderDLLs();
+            SessionOptions options = new SessionOptions();
```

Comment:
>options [](start = 27, length = 7)

SessionOptions is a IDisposable. What happens to this instance when either of the below native methods return an error VerifySuccess throws?

Reply:
Put native methods inside using-block. Since SessionOptions is an IDisposable, dispose function will be called once errors/exceptions happen.

BTW, I found out that our SessionOptions class didn't contain dispose function. Just added.

Reply:
The implementation of SessionOptions is correct. You do not need to change it.

Reply:
I've undo my change.